### PR TITLE
Task 5 heroku

### PR DESCRIPTION
**Pull request by Galcarmi**

### CODE REVIEW

File: express-server/index.js [hunk 1]
```diff
@@ -1,44 +1,52 @@
...
+  require('dotenv').config();
+}
+
 const parentFolder = path.join(__dirname, '../')
-const todoInMemoryDBManager = new TodoInMemoryDBManager();
+const todoMongoDBManager = new TodoMongoDBManager();
 const app = express();
 const port = process.env.PORT || 8000;
 
+const DBURI = `mongodb+srv://admin:${process.env.DBPassword}@cluster0.5zncg.mongodb.net/todoapp?retryWrites=true&w=majority`;
```
Comment (ofirdagan) [inline express-server/index.js:20]: why is this on `index.js`?

File: express-server/DBManager/TodoMongoDBManager.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const { TodoDBManager } = require('./TodoDBManager');
```
Comment (ofirdagan) [inline express-server/DBManager/TodoMongoDBManager.js:1]: why `require`?

File: express-server/DBManager/TodoMongoDBManager.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+const { IdNotFoundError } = require('../errors/IdNotFoundError');
+const { MissingFieldsError } = require('../errors/MissingFieldsError');
+const Todo = require('./models/Todo');
+
+class TodoMongoDBManager extends TodoDBManager {
+  constructor() {
```
Comment (ofirdagan) [inline express-server/DBManager/TodoMongoDBManager.js:7]: no need for empty ctor

File: express-server/DBManager/TodoMongoDBManager.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    super();
+  }
+
+  async addTodo(userId, todo) {
+    if (!todo.content) {
+      throw new MissingFieldsError('content');
+    }
+    
+    const todoToInsert = new Todo({ content: todo.content, userId, isDone: false });
+    const insertedTodo = await todoToInsert.save()
```
Comment (ofirdagan) [inline express-server/DBManager/TodoMongoDBManager.js:17]: `return todoToInsert.save();`

File: express-server/DBManager/TodoMongoDBManager.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    return insertedTodo;
+  }
+
+  async removeTodo(userId, _id) {
+    if (!_id) {
+      throw new MissingFieldsError('id');
+    }
+    
+    const deletedTodo  = await Todo.findOneAndRemove({_id, userId})
```
Comment (ofirdagan) [inline express-server/DBManager/TodoMongoDBManager.js:27]: I don't like the name `Todo` for the mongo model. let's discuss about DAO

File: express-server/DBManager/TodoMongoDBManager.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+    return deletedTodo._doc;
+  }
+
+  async editTodo(userId, todo) {
+    if (!todo._id) {
+      throw new MissingFieldsError('id');
+    }
+
+    let updatedTodo = await Todo.findOneAndUpdate({_id:todo._id, userId}, {...todo});
```
Comment (ofirdagan) [inline express-server/DBManager/TodoMongoDBManager.js:41]: why `let and not `const`

File: express-server/DBManager/TodoMongoDBManager.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    if(!updatedTodo){
+        throw new IdNotFoundError(todo._id);
+    }
+    
+    updatedTodo = {...updatedTodo._doc, ...todo};
```
Comment (ofirdagan) [inline express-server/DBManager/TodoMongoDBManager.js:47]: what's in `updatedTodo._doc`?

File: express-server/DBManager/models/Todo.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const todoSchema = new Schema({
+    userId : String,
```
Comment (ofirdagan) [inline express-server/DBManager/models/Todo.js:5]: do u need to define which field is indexed?

File: express-server/.gitignore [hunk 9]
```diff
@@ -2,3 +2,4 @@
 /dist
 /combined.log
 /error.log
+.env
```
Comment (ofirdagan) [inline express-server/.gitignore:5]: deploy to heruko from root. remove bundles from git.

File: express-server/DBManager/TodoMongoDBManager.js (hunk 2 above)
Comment (Galcarmi) [reply express-server/DBManager/TodoMongoDBManager.js:1]: OLD VERSION

**Pull request merged**